presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PruneCrossJoinColumns.java

```diff
@@ -63,8 +63,6 @@ public Pattern getPattern()
                 .map(ImmutableSet::copyOf)
                 .map(dependencies ->
                         parent.replaceChildren(ImmutableList.of(
-                                joinNode.replaceChildren(ImmutableList.of(
-                                        restrictOutputs(idAllocator, joinNode.getLeft(), dependencies).orElse(joinNode.getLeft()),
-                                        restrictOutputs(idAllocator, joinNode.getRight(), dependencies).orElse(joinNode.getRight()))))));
+                                restrictBinaryChildrenOutputs(idAllocator, joinNode, dependencies, dependencies).get())));
```

Comment:
I thought the rule was to introduce two uses of a function in the commit that introduces that function, to justify its existence?